plugin/lighthouse/handler_test.go

```diff
@@ -658,13 +668,38 @@ func testHeadlessService() {
 				},
 			})
 		})
+		It("should succeed and write an SRV record response", func() {
+			executeTestCase(lh, rec, test.Case{
+				Qname: service1 + "." + namespace1 + ".svc.clusterset.local.",
```

Comment:
Are we sure we are implementing what we wanted?

If I look at the k8s docs I see:

# SRV records
SRV Records are created for named ports that are part of normal or Headless Services.
For each named port, the SRV record would have the form `_my-port-name._my-port-protocol.my-svc.my-namespace.svc.cluster-domain.example`. 

* For a regular service, this resolves to the `port number` and the domain name: `my-svc.my-namespace.svc.cluster-domain.example`. 
* For a headless service, this resolves to multiple answers, one for each pod that is backing the service, and contains the` port number` and `the domain name of the pod of the form auto-generated-name.my-svc.my-namespace.svc.cluster-domain.example.`

Reply:
I added one more test case to cover "_my-port-name._my-port-protocol.my-svc.my-namespace.svc.cluster-domain.example".

We support queries of both port name and protocol and without. When port name and protocol is passed only matching entries are returned. Otherwise, all the ports in the service are returned as individual SRV records as described here 

https://github.com/submariner-io/enhancements/blob/devel/lighthouse/srv-records.md
```
#Cluster IP Service
<portname>.<protocol>.<svc-name>.<svc-namespace>.svc.clusterset.local -> will return the SRV record with the port number and domain name with matching port name and protocol from one of the joined clusters where the service is available. The clusters will be selected based on the Round Robin algorithm.

<portname>.<protocol>.<cluster-name>.<svc-name>.<svc-namespace>.svc.clusterset.local -> will return SRV record from the cluster specified, if available with matching portname and protocol.

If <port-name> and <protocol> is not specified, one SRV record will be returned for each port configured for the service in the selected/specified cluster.

#Headless Service
<portname>.<protocol>.<svc-name>.<svc-namespace>.svc.clusterset.local -> will return the SRV records with port number and domain name for each backend pod with matching portname and protocol in every joined cluster where the service is available.

<portname>.<protocol>.<cluster-name>.<svc-name>.<svc-namespace>.svc.clusterset.local -> will return the SRV records with port number and domain name for each backend pod with matching portname and protocol, in the specified cluster.

If <port-name> and <protocol> is not specified, one SRV record will be returned for each port configured for the service in the selected cluster or all clusters based on the query.

```

Reply:
Thank you, I'm going to link the enhancement proposal to the PR.